pkg/thanos/statefulset.go

```diff
@@ -158,7 +159,37 @@ func makeStatefulSetSpec(tr *monitoringv1.ThanosRuler, config Config, ruleConfig
 		fmt.Sprintf("--eval-interval=%s", tr.Spec.EvaluationInterval),
 		fmt.Sprintf("--tsdb.retention=%s", tr.Spec.Retention),
 	}
-	trEnvVars := []v1.EnvVar{}
+	trEnvVars := []v1.EnvVar{
+		{
+			Name: "POD_NAME",
+			ValueFrom: &v1.EnvVarSource{
+				FieldRef: &v1.ObjectFieldSelector{FieldPath: "metadata.name"},
+			},
+		},
+	}
+
+	// Use defaultReplicaLabelName constant by default if field is missing.
+	// Do not add label if field is set to empty string.
+	replicaLabelName := defaultReplicaLabelName
+	if tr.Spec.ReplicaLabelName != nil {
+		if *tr.Spec.ReplicaLabelName != "" {
+			replicaLabelName = *tr.Spec.ReplicaLabelName
+		} else {
+			replicaLabelName = ""
+		}
+	}
+	// TODO: Users can add other labels to ruler.
+	if replicaLabelName != "" {
+		trCLIArgs = append(trCLIArgs, fmt.Sprintf(`--label=%s="$(POD_NAME)"`, replicaLabelName))
```

Comment:
Do we need to check whether object config is set or not? Since if we add the `thanos_ruler_replica` label by default, then all thanos-query may require to add `--query.replica-label=thanos_ruler_replica` @pgier